src/k8s/pkg/k8sd/database/worker.go

```diff
@@ -20,21 +21,24 @@ var (
 )
 
 // CheckWorkerNodeToken returns true if the specified token can be used to join the specified node on the cluster.
+// CheckWorkerNodeToken will return true if the token is empty or if the token is associated with the specified node
+// and has not expired.
 func CheckWorkerNodeToken(ctx context.Context, tx *sql.Tx, nodeName string, token string) (bool, error) {
 	selectTxStmt, err := cluster.Stmt(tx, workerStmts["select-token"])
 	if err != nil {
 		return false, fmt.Errorf("failed to prepare select statement: %w", err)
 	}
 	var tokenNodeName string
-	if selectTxStmt.QueryRowContext(ctx, token).Scan(&tokenNodeName) == nil {
-		return tokenNodeName == "" || subtle.ConstantTimeCompare([]byte(nodeName), []byte(tokenNodeName)) == 1, nil
+	var expiry time.Time
+	if selectTxStmt.QueryRowContext(ctx, token).Scan(&tokenNodeName, &expiry) == nil {
+		return (tokenNodeName == "" || subtle.ConstantTimeCompare([]byte(nodeName), []byte(tokenNodeName)) == 1) && time.Now().Before(expiry), nil
```

Comment:
```suggestion
        expired :=  (tokenNodeName == "" || subtle.ConstantTimeCompare([]byte(nodeName), []byte(tokenNodeName)) == 1) && time.Now().Before(expiry)
		return expired, nil
```

Reply:
Good point, let's make it even better to read:
```
		isValidToken := subtle.ConstantTimeCompare([]byte(nodeName), []byte(tokenNodeName)) == 1
		notExpired := time.Now().Before(expiry)
		return isValidToken && notExpired, nil
```